### PR TITLE
Rpaasv2 client lib

### PR DESCRIPTION
**Pull request by arthurcgc**
Started implementing rpaasv2 client lib in: rpaas-operator/pkg/rpaas/client

### CODE REVIEW

File: pkg/rpaas/client/client.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+	Description string `json:"description"`
+}
+
+type plan struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Default     bool   `json:"default"`
+}
+
+func (c *RpaasClient) Plans(ctx context.Context, instance *string) ([]plan, error) {
```
Comment (nettoclaudio) [inline pkg/rpaas/client/client.go]: I would named as `GetPlans`...

File: pkg/rpaas/client/client.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+	}
+	return &RpaasClient{
+		httpClient:   &http.Client{},
+		tsuruTarget:  tsuruAPI,
+		tsuruToken:   token,
+		tsuruService: service,
+	}, nil
+}
+
+type flavor struct {
```
Comment (nettoclaudio) [inline pkg/rpaas/client/client.go]: We should export these structs. By the way, we can move them to a types only package, what do you think?

File: pkg/rpaas/client/client.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+	var pathName string
+	switch instance {
+	case nil:
+		pathName = fmt.Sprintf("/resources/plans")
+	default:
+		pathName = fmt.Sprintf("/resources/%s/plans", *instance)
+	}
+
+	req, err := c.newRequest("GET", *instance, pathName, nil)
```
Comment (nettoclaudio) [inline pkg/rpaas/client/client.go]: What does happen here when instance is nil?

File: pkg/rpaas/client/client.go (hunk 2 above)
Comment (arthurcgc) [reply pkg/rpaas/client/client.go]: sure, client.go is already getting big

File: pkg/rpaas/client/client.go (hunk 3 above)
Comment (arthurcgc) [reply pkg/rpaas/client/client.go]: nice catch

File: pkg/rpaas/client/client.go (hunk 1 above)
Comment (arthurcgc) [reply pkg/rpaas/client/client.go]: done

File: pkg/rpaas/client/client.go (hunk 2 above)
Comment (arthurcgc) [reply pkg/rpaas/client/client.go]: i added type/types.go but am using a types.go inside the client package until I learn how to import packages properly in go LOL, apparently I should have put the porject inside my ${GOPATH}

**Pull request merged**